### ISSUE TITLE
Semantic highlighting not working in some cases

### ISSUE THREAD

**Issue by arrufat** [zigtools/zls#1454]
### Zig Version

0.12.0-dev.464+a63a1c5cb

### Zig Language Server Version

20b80784998e342296473a2192ea70935ab84b8d

### Steps to Reproduce

I am using Kakoune with [kak-lsp](https://github.com/kak-lsp/kak-lsp).

Check the zls capabilities:

```
lsp-semantic-tokens:
    types:     [namespace, type, class, enum, interface, struct, typeParameter, parameter, variable, property, enumMember, event, function, method, macro, keyword, modifier, comment, string, number, regexp, operator, decorator, errorTag, builtin, label, keywordLiteral, union, opaque]
    modifiers: [declaration, definition, readonly, static, deprecated, abstract, async, modification, documentation, defaultLibrary, generic, _]
```

I've set all the declarations to appear in bold, and all the members in structs to have an italic variant.

I am using the same config in C++ with clangd, and it works great.

### Expected Behavior

Here's how it should work, and how it works.
- `function`: white → **works**
- `function declaration`: white + bold → **works**
- `method`: white + italic → **doesn't work: no italics**
- `method declaration`: white + italic + bold → **doesn't work: bold but no italics**
- `variable`: blue → **works**
- `variable declaration`: blue + bold → **works**
- `field`: blue + italic → **works**
- `field declaration`: blue + italic + bold → **doesn't work: no bold**

### Actual Behavior

It seems like zls is not distinguishing functions from methods, and also field declarations.

Here's an example in Zig and C++, just to show that my setup works.
![image](https://github.com/zigtools/zls/assets/1671644/9e9e33fb-6651-4277-850b-4687f019d277)

**Comment by Techatrix** [comment]
expected behavior translated into LSP specification:
- container field declaration are missing `TokenModifiers.declaration`.
-  function that can participate in the "dot syntax" should use `TokenType.method`

**Comment by Techatrix** [comment]
does #1455 resolve your issue?

**Comment by arrufat** [comment]
I just tried, and yes, it works perfectly now, thank you :)

**Comment by Techatrix** [comment]
fixed by  #1455
